lib/engine/corporation.rb

```diff
@@ -157,7 +159,11 @@ def president?(player)
     end
 
     def floated?
-      @floated ||= percent_of(self) <= 100 - @float_percent
+      @floatable && (@floated ||= percent_of(self) <= 100 - @float_percent)
```

Comment:
why does this need to check floatable also if it does it should probably check floatable?

Reply:
since it's attr accessor, i would remove the method definition floatable? and just use floatable

Reply:
Have now replaced floatable? with floatable accessor.

The idea of this accessor is to make it possible to sell 10% shares, while still having 20% reserved. The sell of 10% might still exceed floated percent but as long as the 20% certificate is not sold/exchanged the corporation cannot float.

So when the exchange/merge happens 1824 will change floatable to true. (As it is set to false for these Staatsbahn corporation which have presidency certificate reserved for exchange with Coal corporation.)

Reply:
An alternative is to subclass corporation (as 1856 does) but I think it might be confusing for the corporation charter if 50% of the shares is sold and it would show "0% to float" or even "-10% to float"? That could be handled by introducing a game specific method for the presentation.
Have not tried that case though, could do that. Maybe I could handle that case in a graceful way in the presentation?
(e.g. if floated? is false, and percent to float not positive, put in a suitable text)

Reply:
In my new commit I subclassed corporation (and minor) to remove this from general code. Note sure if any other 18xx title uses this so it might be just as well.